src/jdbc/cubrid/jdbc/jci/UStatement.java

```diff
@@ -2278,31 +2331,25 @@ private void read_fetch_data(UInputBuffer inBuffer, UFunctionCode functionCode)
 			fetchedTupleNumber = 0;
 		}
 
-		if (tuples == null) {
-			if (totalTupleNumber > 0) {
-				if (totalTupleNumber > fetchedTupleNumber) {
-					tuples = new UResultTuple[totalTupleNumber];
-				}
-				else {
-					tuples = new UResultTuple[fetchedTupleNumber]; /* for maxRow */
-				}
-			}
-		}
+		tuples = new UResultTuple[fetchedTupleNumber];
 
 		for (int i = 0; i < fetchedTupleNumber; i++) {
-			readATuple(i + readTupleNumber, inBuffer);
+			readATuple(i, inBuffer);
 		}
-		readTupleNumber += fetchedTupleNumber;
 
-			if (functionCode == UFunctionCode.GET_GENERATED_KEYS) {
-				isFetchCompleted = true;
-			}
+		if (functionCode == UFunctionCode.GET_GENERATED_KEYS) {
+			isFetchCompleted = true;
+		}
 
 		if (functionCode == UFunctionCode.FETCH
 		        && relatedConnection
 		                .protoVersionIsAbove(UConnection.PROTOCOL_V5)) {
 			isFetchCompleted = inBuffer.readByte() == 1 ? true : false;
 		}
+		
+		if (stmt_cache_data != null && totalFetchSize < relatedConnection.getClientCacheSize()) {
+			stmt_cache_data.setCacheData(tuples, currentFirstCursor, fetchedTupleNumber);
+		}
```

Comment:
if totalFetchSize exceeds ClientCacheSize, the cache needs to be cleared.
```
if (stmt_cache_data != null)
{
  if (totalFetchSize < relatedConnection.getClientCacheSize())
  {
    // add cache data
  }
  else
  {
    // clear cache
    stmt_cache_data = null;
  }
}
```

Reply:
for exceed case, no more cache? I thought that if the cache exceeds max size then the only remain results not cached.

Reply:
By discussion, there is no problem for partially cached data. it does not need to clear the cache for exceeding ClientCacheSize case.